core/src/main/scala/org/apache/spark/ui/exec/ExecutorsPage.scala

```diff
@@ -69,13 +73,13 @@ private[ui] class ExecutorsPage(
     }
 
     val execInfo = activeExecutorInfo ++ deadExecutorInfo
-    val execInfoSorted = execInfo.sortBy(_.id)
+    val execInfoSorted = execInfo.sortWith((a, b) => strToInt(a.id) > strToInt(b.id))
```

Comment:
I think I'd use `Ordering` trait instead as it's not very easy to spot what the ordering is.

```
case class Info(id: String)
val infos = Seq(Info("5"), Info("driver"), Info("1"), Info("10"), Info("500"))

implicit val onString = Ordering[Int].on((s: String) => util.Try(s.toInt).getOrElse(-1))
infos.sortBy(_.id)
```